### ISSUE TITLE
ghcjs-boot failing on Mac OS X 10.11

### ISSUE THREAD

**Issue by submain** [ghcjs/ghcjs#433]
While trying to execute ghcjs-boot under Mac OS X 10.11 I get:

    [...]
    Resolving dependencies...
    Configuring integer-gmp-0.5.1.0...
    configure: WARNING: unrecognized options: --with-compiler, --with-hc-pkg, --with-gcc
    checking build system type... x86_64-apple-darwin15.0.0
    checking host system type... x86_64-apple-darwin15.0.0
    checking target system type... x86_64-apple-darwin15.0.0
    checking for gcc... gcc
    checking whether the C compiler works... no
    configure: error: in `/Users/user/.ghcjs/x86_64-darwin-0.2.0-7.10.2/ghcjs/ghcjs-boot/boot/integer-gmp':
    configure: error: C compiler cannot create executables
    See `config.log' for more details
    Failed to install integer-gmp-0.5.1.0
    cabal: Error: some packages failed to install:
    array-0.5.1.0 depends on integer-gmp-0.5.1.0 which failed to install.
    base-4.8.0.0 depends on integer-gmp-0.5.1.0 which failed to install.
    binary-0.7.5.0 depends on integer-gmp-0.5.1.0 which failed to install.
    bytestring-0.10.6.0 depends on integer-gmp-0.5.1.0 which failed to install.
    containers-0.5.6.3 depends on integer-gmp-0.5.1.0 which failed to install.
    deepseq-1.4.1.1 depends on integer-gmp-0.5.1.0 which failed to install.
    integer-gmp-0.5.1.0 failed during the configure step. The exception was:
    ExitFailure 77
    pretty-1.1.3.2 depends on integer-gmp-0.5.1.0 which failed to install.
    primitive-0.6 depends on integer-gmp-0.5.1.0 which failed to install.
    template-haskell-2.10.0.0 depends on integer-gmp-0.5.1.0 which failed to
    install.
    transformers-0.4.3.0 depends on integer-gmp-0.5.1.0 which failed to install.
    ghcjs-boot: error running: /usr/local/bin/cabal --config-file /Users/user/.ghcjs/x86_64-darwin-0.2.0-7.10.2/ghcjs/cabalBootConfig --ignore-sandbox install ./boot/array ./boot/base ./boot/binary ./boot/bytestring ./boot/containers ./boot/deepseq ./boot/integer-gmp ./boot/pretty ./boot/primitive ./boot/template-haskell ./boot/transformers --solver=topdown --global --ghcjs --one-shot --avoid-reinstalls --builddir dist --with-compiler /Users/user/Projects/reflex-sandbox/.cabal-sandbox/bin/ghcjs --with-hc-pkg /Users/user/Projects/reflex-sandbox/.cabal-sandbox/bin/ghcjs-pkg --prefix /Users/user/.ghcjs/x86_64-darwin-0.2.0-7.10.2/ghcjs --enable-documentation --haddock-html --haddock-hyperlink-source --enable-shared --enable-library-profiling --root-cmd /Users/user/.ghcjs/x86_64-darwin-0.2.0-7.10.2/ghcjs/run.sh
    exit status: 1
    stderr: Warning: No remote package servers have been specified. Usually you would have
    one specified in the config file.
    configure: WARNING: unrecognized options: --with-compiler, --with-hc-pkg, --with-gcc
    configure: error: in `/Users/user/.ghcjs/x86_64-darwin-0.2.0-7.10.2/ghcjs/ghcjs-boot/boot/integer-gmp':
    configure: error: C compiler cannot create executables
    See `config.log' for more details
    cabal: Error: some packages failed to install:
    array-0.5.1.0 depends on integer-gmp-0.5.1.0 which failed to install.
    base-4.8.0.0 depends on integer-gmp-0.5.1.0 which failed to install.
    binary-0.7.5.0 depends on integer-gmp-0.5.1.0 which failed to install.
    bytestring-0.10.6.0 depends on integer-gmp-0.5.1.0 which failed to install.
    containers-0.5.6.3 depends on integer-gmp-0.5.1.0 which failed to install.
    deepseq-1.4.1.1 depends on integer-gmp-0.5.1.0 which failed to install.
    integer-gmp-0.5.1.0 failed during the configure step. The exception was:
    ExitFailure 77
    pretty-1.1.3.2 depends on integer-gmp-0.5.1.0 which failed to install.
    primitive-0.6 depends on integer-gmp-0.5.1.0 which failed to install.
    template-haskell-2.10.0.0 depends on integer-gmp-0.5.1.0 which failed to
    install.
    transformers-0.4.3.0 depends on integer-gmp-0.5.1.0 which failed to install.

When checking the config.log, I see that it cannot find the standard C includes.

I believe this is because the new versions of OS X, namely xcode, places includes and standard libraries in unusual locations. 

Is there a way to adjust the CCFLAGS and/or include directories used in the build process? 

I'd be willing to solve this issue and post the fix, but I am not familiar on what's the best way to change the library and include paths of ghcjs-boot.

**Comment by luite** [comment]
does installing packages that have a configure script with GHC still work? I think we should try to keep the way system libraries are found as close to that as possible.

**Comment by submain** [comment]
Yes - installing integer-gmp >= 1.0 works, if that's what you mean. Though I'm not sure if the lookup paths have been changed since the gmp that is included in ghcjs-boot (integer-gmp-0.5).

**Comment by luite** [comment]
Ok, I'll look into this. Unfortunately I haven't upgraded to 10.11 yet, and I'd like to wait a little longer before switching. Do you know if there's an easy way to set it up in a VM?

**Comment by submain** [comment]
Are you still on 10.9 or below? I think the build system has changed starting from 10.10. There is a way to run it in virtualbox: https://www.youtube.com/watch?v=tafqhSUfKnY

I think that should give roughly the same result as 10.11.

**Comment by luite** [comment]
I'm on 10.10, and it seems to boot ok here (it's been my main dev system), so I'm not quite sure what's going on.

**Comment by submain** [comment]
Hmmm, weird. What is your xcode version?

**Comment by luite** [comment]
I've just updated to the latest, but haven't rebooted ghcjs with that yet

**Comment by submain** [comment]
Ok, I don't know what else it could be. Please let me know if the new one boots for you.

**Comment by alios** [comment]
Hi! I've just done a fresh install using ghcjs from git. using most recent xcode on most recent osx el captian. I'am using ghc 7.10.2 installed with ```brew install ghc``` for compilation.

The only thing i noticed after updating to most recent xcode was (haven't it started yet) that fireing up a ghci in the terminal i got this: https://twitter.com/alios/status/657612448829083648

I guess ghci calls some toolchain things at runtime (linker?) ... after accepting that license stuff (with sudo ?!?!?) everthing was working as expected again.

**Comment by submain** [comment]
Yeah I got that weird license stuff too. Though I am using haskell platform instead of the brew one. I'll remove my haskell and install it again with the one in brew and see if that solves the problem.

**Comment by submain** [comment]
It works! It turns out that the new xcode needs an extra step to install its binaries. Once I ran

    xcode-select --install

Then it installed the libraries in the correct places and ghcjs was able to boot. Thanks a lot for your help!

**Comment by luite** [comment]
I have things working properly with the latest xcode too now, but I get linker warnings all the time now that my object files are for OS X 10.11 while I'm still on 10.10 :( . Oh well, at least it's not completely broken.

**Comment by submain** [comment]
I get the same warnings too. Though it seems to work. Still struggling with building reflex-dom though, but that's for a different thread.

**Comment by alios** [comment]
@submain me too :) i think it needs some work to be compilable with most recent ghcjs and esp. ghcjs-dom. tried a cuple of branches - is there already another thread somewhere?
